### PR TITLE
WIP: filter_replace_character: Add plugin to replace a generic character in key names

### LINKED ISSUE #1305: Any "Replace_Dots On" option for Kafka output？
I'm using fluent-bit to send logs to kafka, and ES is the consumer

Facing the error with kubernetes labels 

```
[43]: index [logx-k8s_2019-04-25], type [logx-k8s], id [5X28U2oB2gDMRtJZMcad], message [ElasticsearchException[Elasticsearch exception [type=mapper_parsing_exception, reason=Could not dynamically add mapping for field [app.kubernetes.io/instance]. Existing mapping for [kubernetes.labels.app] must be of type object but found [text].]]] (org.nesc.ecbd.sink.task.ElasticSearchSinkTask)
```

Found ES has 

`Replace_Dots On` options but cannot find any in Kafka output

Is there any possibility that Kafka can have the same solution as ES

or any other alternative way ?

Thanks a lot 

reference :

https://github.com/fluent/fluent-bit/issues/708

### LINKED ISSUE #1538: Kubernetes Filter - Add "Replace_Dots" function found in Elasticsearch Filter
**Is your feature request related to a problem? Please describe.**
We currently have fluentbit tailing and parsing logs, but we subsequently pass these into fluentd for the benefit of buffering logs. 

Due to this, we're using the Kubernetes filter, but not using the Elasticsearch output plugin that comes with `Replace_Dots` feature.

As this is missing, we're experiencing the same problem found in the original ticket raised for the `Replace_Dots` feature: https://github.com/fluent/fluent-bit/issues/708

Example conflict error from ES: 

```
2019-08-29 08:03:47 +0000 [warn]: #0 dump an error event: error_class=Fluent::Plugin::ElasticsearchErrorHandler::ElasticsearchError error="400 - Rejected by Elasticsearch [error type]: mapper_parsing_exception [reason]: 'Could not dynamically add mapping for field [kubernetes_labels_app.kubernetes.io/instance]. Existing mapping for [kubernetes_labels_app] must be of type object but found [text].'" location=nil tag="kube.tag.log" time=2019-08-29 08:03:39.488097120 +0000 record={"log"=>" LOG"}
```

**Describe the solution you'd like**
It would be great if the Kubernetes either automatically replaced the dots, or offered the same option as the Elasticsearch filter. 

**Describe alternatives you've considered**
See above

Alternatively, the suggested "generic" filter suggested here would work: https://github.com/fluent/fluent-bit/issues/1305

**Additional context**
Right now, a lot of our logs are rejected due to this, I can make some customized templating changes and a number of other complicated solutions, but having this option would solve the problem much faster.

### PR DESCRIPTION
**Pull request by PettitWesley**
Fixes #1538
Fixes #1305 

Related to:
- #708
- #1279

### CODE REVIEW

**Comment by PettitWesley** [comment]
I probably will not be able to work on this for a while- if anyone else wants to take a stab a finishing it, feel free to.

File: plugins/filter_replace_character/replace.c [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+            modify = FLB_FALSE;
+            key = (kv+i)->key
+            if (k->type == MSGPACK_OBJECT_BIN) {
+                key_str  = (char *) k->via.bin.ptr;
+                key_str_size = k->via.bin.size;
+                modify = FLB_TRUE;
+            }
+            else if (k->type == MSGPACK_OBJECT_STR) {
+                key_str  = (char *) k->via.str.ptr;
+                key_str_size = k->via.str.size;
```
Comment (PettitWesley) [inline plugins/filter_replace_character/replace.c]: I think `k` is supposed to be `key`

**Comment by edsiper** [comment]
@PettitWesley is this still in WIP or ready to be reviewed for a possible merge ?

**Comment by PettitWesley** [comment]
@edsiper This code is not ready for review; it does not work. Right now I don't have the bandwidth to finish it.

#2125 Was supposed to complete it.

**Comment by edsiper** [comment]
closing this one for now.

**Pull request closed without merging**